hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancer.java

```diff
@@ -134,11 +142,17 @@ public void unBalancedNodesListShouldBeEmptyWhenClusterIsBalanced() {
   @Test
   public void containerBalancerShouldStopWhenMaxDatanodesToBalanceIsReached() {
     balancerConfiguration.setMaxDatanodesToBalance(2);
-    balancerConfiguration.setThreshold(0);
+    balancerConfiguration.setThreshold(1);
```

Comment:
Why is `threshold` set to `1` here? A threshold of 1 means balancing is not required, which is incorrect for this test since we are checking the `maxDatanodesToBalance` configuration. Moreover threshold belongs to `[0, 1)`.